mkdocs/tests/nav_tests.py

```diff
@@ -610,3 +610,125 @@ def test_indented_toc_missing_child_title(self):
         self.assertEqual(str(site_navigation).strip(), expected)
         self.assertEqual(len(site_navigation.nav_items), 3)
         self.assertEqual(len(site_navigation.pages), 6)
+
+    def test_edit_uri_simple(self):
+        """
+        Ensure that set_edit_url creates well formed URLs for simple edit_uri
+        """
+
+        pages = [
+            'index.md',
+            'internal.md',
+            'sub/internal.md',
+            'sub1/sub2/internal.md',
+        ]
+
+        repo_url = 'http://example.com/'
+        edit_uri = 'edit/master/docs/'
+
+        site_navigation = nav.SiteNavigation(pages)
+
+        expected_results = (
+            repo_url + edit_uri + pages[0],
+            repo_url + edit_uri + pages[1],
+            repo_url + edit_uri + pages[2],
+            repo_url + edit_uri + pages[3],
+        )
+
+        for idx, page in enumerate(site_navigation.walk_pages()):
+            page.set_edit_url(repo_url, edit_uri)
+            self.assertEqual(page.edit_url, expected_results[idx])
+
+        # Ensure the '/' is added to the repo_url for simple edit_uri
+        repo_url = 'http://example.com'
+        edit_uri = 'edit/master/docs/'
```

Comment:
Sure, I will include those tests later today. I think I'll just create 2 test functions, one containing all tests for posix-style paths and the other one for windows-style paths, to reduce the length of the code.